app/services/samples/metadata/fields/edit_service.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    module Fields
+      # Service used to validate the edit_fields param and construct the metadata update param to be passed
+      # to metadata_controller#update and Samples::Metadata::UpdateService
+      class EditService < BaseService
+        SampleMetadataFieldsEditError = Class.new(StandardError)
+        attr_accessor :project, :sample, :key_edit, :value_edit, :metadata_update_params
+
+        def initialize(project, sample, user = nil, params = {})
+          super(user, params)
+          @project = project
+          @sample = sample
+          @key_edit = params['edit_field']['key']
+          @value_edit = params['edit_field']['value']
+          @metadata_update_params = {}
+        end
+
+        def execute
+          authorize! @project, to: :update_sample?
+
+          validate_sample_in_project
+
+          validate_edit_fields
+
+          construct_metadata_update_params
```

Comment:
**nitpick (blocking):** after crafting the update params this service should call the `Samples::Metadata::UpdateService`

Reply:
Updated in 5df4af3ff5328969250bf3de5a14d6bde116f993